lewis/devices/julabo/interfaces/julabo_stream_interface_2.py

```diff
@@ -0,0 +1,58 @@
+# -*- coding: utf-8 -*-
+# *********************************************************************
+# plankton - a library for creating hardware device simulators
+# Copyright (C) 2016 European Spallation Source ERIC
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+# *********************************************************************
+
+from lewis.adapters.stream import StreamAdapter, Cmd
+
+
+class JulaboStreamInterfaceV2(StreamAdapter):
+    """Julabos can have different commands sets depending on the version number of the hardware.
+
+    This protocol matches that for: FP50-HE (unconfirmed).
+    """
+
+    protocol = "julabo-version-2"
+
+    commands = {
+        Cmd('get_bath_temperature', '^IN_PV_00$'),
+        Cmd('get_external_temperature', '^IN_PV_01$'),
+        Cmd('get_power', '^IN_PV_02$'),
+        Cmd('get_set_point', '^IN_SP_00$'),
+        Cmd('get_high_limit', '^IN_SP_03$'),    # Different from version 1
```

Comment:
Thanks for your feedback and especially for that use case - we didn't think of something like `Var` before. One thing that we also discovered in our discussions is for the case of trivial getter and non-trivial setter, where you can define both `Var` and `Cmd`, for example like this:

```python
class Interface(StreamAdapter):
    commands = {
        Var('param', read_pattern='^P$', doc='Some param')
        Cmd('set_param', pattern='^P=([0-9]+)$, argument_mappings=(int,))
    }

    param = 10

    @check_limits(0, 20)
    def set_param(self, new_param):
        """Set param after checking limits."""
        self.param = new_param
```

Then you get a trivial getter and a limit checked, type converted setter.